viewer/packages/sector-parser/visual-tests/GeneralCylinder.VisualTest.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+import * as THREE from 'three';
+import {
+  createPrimitiveInterleavedGeometry,
+  GeneralCylinder,
+  PrimitiveName
+} from '../../../test-utilities/src/primitives';
+import { SimpleTestFixtureComponents, SimpleVisualTestFixture } from '../../../visual-tests';
+import { setConeGeometry } from '../src/reveal-glb-parser/primitiveGeometries';
+import { RevealGeometryCollectionType } from '../src/types';
+import { getMaterialsMap } from './testMaterials';
+
+export default class GeneralCylinderVisualTest extends SimpleVisualTestFixture {
+  public setup(simpleTestFixtureComponents: SimpleTestFixtureComponents): Promise<void> {
+    const { camera, scene } = simpleTestFixtureComponents;
+
+    const generalCylinderDescriptor = {
+      treeIndex: 0,
+      color: [0, 0, 0, 1],
+      centerA: [0, 0, 1.4663076400756836],
+      centerB: [0, 0, -1],
+      radius: 0.2,
+      planeA: [0.4161977469921112, 0.07338689267635345, 0.9063078165054321, 2],
+      planeB: [0, 0, -1, 0],
+      localXAxis: [1, 0, 0],
+      angle: 0,
+      arcAngle: Math.PI * 2
+    } as GeneralCylinder;
+
+    const group = this.initializeGroup(scene);
+    const geometry = createPrimitiveInterleavedGeometry(PrimitiveName.GeneralCylinder, [generalCylinderDescriptor]);
+    setConeGeometry(geometry);
+    const materials = getMaterialsMap(1);
+
+    const generalCylinderMaterial = materials.get(RevealGeometryCollectionType.GeneralCylinderCollection)!;
+    const mesh = new THREE.Mesh(geometry, generalCylinderMaterial);
+    mesh.frustumCulled = false;
+    mesh.onBeforeRender = () => {
+      (generalCylinderMaterial.uniforms.inverseModelMatrix?.value as THREE.Matrix4)?.copy(mesh.matrixWorld).invert();
+      (generalCylinderMaterial.uniforms.cameraPosition?.value as THREE.Vector3)?.copy(camera.position);
+    };
+    group.add(mesh);
+
+    camera.position.set(0, 0, -3);
```

Comment:
I notice in the snapshot image that the top of the cylinder isn't visible - I guess it might be more useful if the top was seen as well?